app/dataflow/middlewares/onStoreChange.js

```diff
@@ -0,0 +1,60 @@
+import { LOCATION_CHANGE } from 'react-router-redux';
+import types from '../actions/types';
+import { getWeather, getForecast, changeCity, setMetric, redirectToCity, getNearestTo } from '../actions/index';
+
+const storeChangeHandler = (prevState, newState, action, dispatch) => {
```

Comment:
@AlexeyBarkow you can separate all this cases as minimum to 2 different. Read about reducer composition.